otx/cli/tools/train.py

```diff
@@ -144,15 +135,50 @@ def parse_args():
         help="Total number of workers in a worker group.",
     )
 
-    add_hyper_parameters_sub_parser(parser, hyper_parameters)
-
-    return parser.parse_args(), template, hyper_parameters
+    return parser
 
 
-def main():  # pylint: disable=too-many-branches
+# pylint: disable=too-many-branches, too-many-statements
+def main():
     """Main function that is used for model training."""
-    # Dynamically create an argument parser based on override parameters.
-    args, template, hyper_parameters = parse_args()
+    parser = get_parser()
+
+    args, _ = parser.parse_known_args()
+    train_workspace_path = os.getcwd()
+
+    has_template_yaml = Path(args.template).exists()
+    has_data_yaml = Path(args.data).exists()
+    was_built_before = has_template_yaml and has_data_yaml
+    if not was_built_before:
+        # Prepare build
+        train_workspace_path = os.path.join(args.save_model_to)
+        template = find_and_parse_model_template(args.template) if has_template_yaml else None
+
+        # Build
+        builder = Builder()
+        build(
+            builder=builder,
+            train_data_roots=args.train_data_roots,
+            val_data_roots=args.val_data_roots,
+            workspace_root=train_workspace_path,
+            otx_root=get_otx_root_path(),
+            template=template,
+        )
+
+    # Update configurations made by builder
+    # When users input template argument, need to overwrite
+    if args.template != "./template.yaml":
+        template = find_and_parse_model_template(args.template)
+    else:
+        template = find_and_parse_model_template(os.path.join(train_workspace_path, "template.yaml"))
+    args.data = os.path.join(train_workspace_path, "data.yaml")
+    data_config = configure_dataset(args)
+
+    # Get hyper parameters schema.
+    hyper_parameters = template.hyper_parameters.data
+    assert hyper_parameters
```

Comment:
From what I remember, we decided to avoid using asserts as much as possible. How about raising issue?

Reply:
Sure, no problem. I just used previous implemented code [line62](https://github.com/openvinotoolkit/training_extensions/pull/1592/files#diff-d4366781a54d3bbd5d4c49115d58e31a837715214588d4409e388a359bbf2b9eL62)